cloudinfo/containers.go

```diff
@@ -47,3 +49,68 @@ func (infoSvc *CloudInfoService) GetClusterConfigPath(clusterId string, basePath
 
 	return configDetails.FilePath, nil
 }
+
+// GetAlbState retrieves the details of State corresponding to ALB ID
+// albId: the ID of the ALB
+// Returns the State for an ALB ID.
+func (infoSvc *CloudInfoService) GetAlbState(albId string) (status string, err error) {
+	albConfig, detailedResponse, err := infoSvc.albService.GetClusterALB(infoSvc.albService.NewGetClusterALBOptions(albId))
+	if err != nil {
+		log.Println("Failed to get Cluster ALB details for ", albId, ":", err, "Full Response:", detailedResponse)
+		return "", err
+	}
+
+	// If any specific operation to perform for a state(healthy, critical, pending) is requried.
+	/*	if *albConfig.State == "healthy" {
+		} else if *albConfig.State == "critical" {
+		} else {
+		}
+	*/
+	return *albConfig.State, nil
+}
+
+// GetAlbIds retrieves the list of all ALBs in a cluster
+// clusterId: the ID or name of the cluster
+// Returns a list all ALB IDs in a cluster. If no ALB IDs are returned, then the cluster does not have a portable subnet.
+func (infoSvc *CloudInfoService) GetAlbIds(clusterId string) (ids []string, err error) {
+	clusterAlbs, detailedResponse, err := infoSvc.albService.GetClusterALBs(infoSvc.albService.NewGetClusterALBsOptions(clusterId))
+	if err != nil {
+		log.Println("Failed to get ALB IDs for ", clusterId, ":", err, "Full Response:", detailedResponse)
+		return []string{}, err
+	}
+
+	// don't bother looping if empty
+	if len(clusterAlbs) == 0 {
+		return
```

Comment:
you might want to be specific with this return, and return an empty array and `nil` for the error. I'm not sure what gets returned by default if you do not specify. something like `return [], nil` would be better than leaving empty